### ISSUE TITLE
[Zero configuration] Add possibility to build a solution with *.nuproj projects

### ISSUE THREAD

**Issue by AndirNotes** [terrajobst/nuproj#80]
It would be handy to allow build an existing solution *.sln with *.nuproj projects without any additional build configuration (like [Checked-in Toolkit](https://github.com/terrajobst/nuproj/blob/master/docs/Build.md)). 
I think it could be done via automatic restore NuProj package during build *.nuproj project and using it as default location of NuProj MSBuild targets and tasks.

**Comment by terrajobst** [comment]
Building would work; what wouldn't work today is the VS integration, i.e. the project would show up as not being loaded. So building inside the IDE would not build .nuproj either. However, command line builds would work. @AArnott had some ideas on how to make this work.

Also, to be clear, the title says 'Checked-in Toolkit' but it doesn't actually require to be checked in. You can restore NuProj build tools and targets via a NuGet package already.

**Comment by AndirNotes** [comment]
I suppose that NuProj installed in MsBuild extension path. And now if I try to build *.nuproj (msbuild.exe SomeProject.nuproj) - I will get error 'error MSB4 057: The target "_NuProjGetProjectClosure" does not exist in the project.'.
The same issue if I have restored NuProj package and set property NuProjPath only. 
Build from command line works only if I set additional property CustomAfterMicrosoftCommonTargets=$(NuProjPath)Microsoft.Common.NuProj.targets;
It's quite inconvenient.

**Comment by AArnott** [comment]
I think NuProj could load in VS just fine without the VSIX installed, so long as the .targets and .xaml files are brought down in the NuGet package. Yes, this would require that NuGet Restore be executed manually from the command line since it won't happen in the IDE as that requires the project to load first (chicken and egg problem).
I'd love to see all the build dependencies be included in the NuGet package and wanted to spend some time in the near future making that happen.

**Comment by kovalikp** [comment]
Do the rules and VS specific targets need to be in the main .targets file? If not, how about put them into separate file and import them conditionally from `$(MSBuildExtensionsPath)`?

**Comment by AArnott** [comment]
No, they don't have to be in the main .targets file. But I don't want *anything* being imported from `$(MSBuildExtensionsPath)`. It should all come from NuGet otherwise the design-time experience in VS is broken unless the MSI is installed.

**Comment by jjcaraballo01** [comment]
@AndirNotes Did you ever come up with a resolution for the "_NuProjGetProjectClosure" issue you were seeing? I'm experiencing the same symptoms and have been unable to get things to work by adding the additional CustomAfterMicrosoftCommonTargets property. Were you strictly setting the property through command line, or can you do this in the .nuproj file. I'm trying to figure this out for a larger MSBuild / CI Integration.

**Comment by AArnott** [comment]
@kovalikp has written a nuget package that only adds the small .targets file that defines this target. that seems like a workable solution to me.
